### DISCUSSION TITLE
コントリビューターが30人以上になった！

### DISCUSSION THREAD

**Discussion by kazzyfrog** [first-contributions-ja/first-contributions-ja.github.io#248] [12: 新機能の提案]
## 概要
このスレを立てた現時点で、
Contributors.json に記載されている、コントリビューターのリストが30人です！

当初、
**（とりあえず）トップページに表示できる絵文字を30個（１セクション10個*現在３セクション）**
として、実装していました。

なので、これ以降のコントリビュートによって、
配列の先頭からサイト上に表示されなくなります！

これに関するアイデアを、ゆるく書いていきます👋

## その他
上記以外にも、さらなる改善点、提案があれば、自由に書き込んでOKです！

**Comment by kazzyfrog** [comment]
アイデア 1：
「Thank you for 30+ Contributors 的なセクション」を追加して、
もう１０個絵文字を表示できるようにする

**Reply by kazzyfrog** [reply]
あ、それか前みたいに、現在のコントリビューターの人数を表示でもいいのか！
でっかい文字で、
「Thank you for [N] Contributors 🎉」
的な！

**Reply by pss-aileen** [reply]
セクション追加して、人数表示いいと思います！

↑これぜひ取り入れたいです！

**Reply by kazzyfrog** [reply]
ただ懸念点としては、
１セクション10人なので、少し延命しただけで、また同じ問題が発生しそうかもです！
(コントリビューターのリストが40人を超えた時。。)

ただ、それとは別で、もう１セクション追加して人数表示があっても良さそうですよね！

**Reply by kazzyfrog** [reply]
とりあえず、これでいいかもですね！
- セクション追加して、コントリビューターの人数を表示する
- 40人を超える場合の対処は、またあとで考える！

問題なければ、
新たなセクションの作成のイシューを立てて、そのイシュー内で、
- 仮デザイン？を作成
- 実装

上の流れで、余裕がある時に少しずつやっていきます🙌

**Reply by pss-aileen** [reply]
はい、こちら問題ないと思います！
すみませんがよろしくお願いいたします🙇

**Reply by kazzyfrog** [reply]
めっちゃ仮ですが、
ざっくりこんな感じでどうかなーというの、一旦共有しておきます！

<img width="1016" alt="スクリーンショット 2024-07-14 0 11 26" src="https://github.com/user-attachments/assets/24d71a20-7568-4814-8dda-dfd989fb3d2f">

**Reply by kazzyfrog** [reply]
その他：
- 絵文字は、背景に合わせてNoto Emoji にしようかなと！
- 下のテキストはまだ仮で、微妙に変える可能性が高いです！
- 1番下のセクションとして追加する。

こんな感じで問題なければ、とりあえず実装して近日中にPR出しますよ！

**Reply by pss-aileen** [reply]
すごくいいと思います！！ありがとうございます！✨
個人的にはこちらでOKです！！

**Reply by kazzyfrog** [reply]
ありがとうございます👍
→ #252

**Comment by pss-aileen** [comment]
あと、このDiscussionで良いかわかりませんが、1つ気になっていたことがありました。

プルリクが続けてきて、いくつかを連続でマージした場合、自分が選んだ色が反映されている状態を確認できないのではないか？という点です。

個人的には自分が変更を加えて反映されているのを確認するのはすごく嬉しいというか、楽しいかなと思っているので、せっかくコントリビュートしていただいたからには見てほしい！！といった気持ちがあります。

実際コントリビュートしていただいた方に意見を伺えるのが一番いいのですが...。（もし誰か見てくださっていたら教えてください...！！！）

もうちょっと「貢献した感」「自分の内容がページに反映されている感」を出したいなと思っています。
具体的にまとまっているわけではありませんが、以下のような機能があれば、↑の悩みを少し解決できるような気がしています。

- トグルボタンを押すと、全部の絵文字が裏返り、ユーザーアイコンが表示された状態になる
- 各ユーザーのアイコンを押すと、その人の色が背景色に適応される

みなさんどう思いますか...？

**Reply by pss-aileen** [reply]
あ、あと、日本感がほしくメインカラーを赤としてヘッダーフッターなどに使っていましたが、これらも全部色変わればいいな...なんて思ってます...

**Reply by kazzyfrog** [reply]
> プルリクが続けてきて、いくつかを連続でマージした場合、自分が選んだ色が反映されている状態を確認できないのではないか？という点

これは自分も思ってました！

できるだけ、数時間の間隔を空けるとかで工夫してましたが、確かに何かもっといい工夫できたら良さそうですね！

**Reply by kazzyfrog** [reply]
> 日本感がほしくメインカラーを赤としてヘッダーフッターなどに使っていましたが、これらも全部色変わればいいな...なんて思ってます...

これも、確かに赤と合わない？ケースもあると思うので、全部変える方向で揃えてもいいかもですね！

**Comment by kazzyfrog** [comment]
あと、余裕があれば、（気が向いた時にでも）
1番下のセクションの絵文字の表示を、もう少しはっきりさせてもいいかもです！

優先度は低そうですが、
色によっては、絵文字があるのかどうかわからない時があるので！

**Reply by pss-aileen** [reply]
最近 Noto Color Emoji があることを知って、そちらを使っても良いのかなと思い始めました。

ただ背景と馴染ませるとか、1番目のセクションのテキストが見づらくなるので対策は必要かなとは思うのですが、色付き絵文字にしたら、一番下のセクションでも存在感があるのでいいかな...！と思いました。

**Reply by kazzyfrog** [reply]
確かにいいかもですね！

確かに画面上が騒がし過ぎ？になるかもですが、
表示確認してみて背景と調整してみるのもいいかもです！

何より、薄い色でも、はっきり絵文字が見えそうです！

**Reply by kazzyfrog** [reply]
ちなみに、Noto Color Emojiに置き換えてみたら、こんな感じです！
現在のまま、透明度0.5とかだと、
カラーでも、意外とちょうどいいかもです！

問題なければ、合わせて変更しておきますよ！👍
![localhost_3000_ (4)](https://github.com/user-attachments/assets/132f8bac-f200-44e0-ad56-b032823f8a61)

**Reply by pss-aileen** [reply]
Noto Color Emojiありがとうございます！！
下の方も見えますね！
こちらで良いと思います✨🙆‍♀️

**Comment by kazzyfrog** [comment]
関連して、これも優先度低いですが、改善点として落としておきます！

## 現状
- ユーザーがチュートリアルを完了した時に、Prettierによるフォーマットが行われていない！
→完全にうっかりしてました。。
→Huskyは、npmパッケージなので、当然ユーザーが`npm install`しない場合は動作しない。
→なので、Prettierによるフォーマットが行われてないですね

## 解決策
GitHub Actions で、GitHub上でpush時などに、`npm format`コマンドを実行し、コミットする！
→これで、GitHub上で強制的に（ユーザーに負担をかけず）Prettierを走らせることができるかと思います！

## 参考：
#250

**Reply by pss-aileen** [reply]
直近のプルリクで、Contributions.json でインデントがなかったのですが、どうしようか迷った結果、そのままマージしてしまったので着手したいなと思いました。ちょっとこちら始めてなのでやり方調査してみます。

**Reply by kazzyfrog** [reply]
了解です！

新たに、Actions のワークフローファイルを作成し、
その中で、PRのタイミングなどで prettier（ついでにESLintも一緒に実行しても良いかも）の実行・書き込みを行う感じですかね！

**Reply by pss-aileen** [reply]
こちら私やってみてもいいですか？ちょっとお時間いただくかもしれませんが...！やりたいです！

> PRのタイミングなどで prettier（ついでにESLintも一緒に実行しても良いかも）の実行・書き込みを行う感じですかね！

PRのタイミングで走らせることもできるんですね！試してみます！

**Reply by kazzyfrog** [reply]
ぜひぜひ！🙌
お手隙の際で大丈夫ですよ！

Actionsのワークフローを[発火させるイベント](https://docs.github.com/ja/actions/using-workflows/events-that-trigger-workflows)めっちゃあるので、今回だとPRのタイミングとかが良いかもですね！

よろしくお願いしますー！
